runtime/src/main/java/io/quarkiverse/githubapp/GitHubConfigFileProvider.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkiverse.githubapp;
+
+import java.util.Optional;
+
+import org.kohsuke.github.GHRepository;
+
+/**
+ * A provider of configuration files from {@link org.kohsuke.github.GHRepository GitHub repositories}.
+ * <p>
+ * The file is extracted from
+ * <p>
+ * Inject as a CDI bean.
+ * <p>
+ * <strong>NOTE:</strong> You generally will not need this bean when processing events,
+ * as configuration files can be automatically injected into event listener methods,
+ * simply by annotating a parameter with {@link ConfigFile}.
+ * This provider is mostly useful for non-event use cases (e.g. cron jobs).
+ *
+ * @see ConfigFile
+ * @see ConfigFile.Source
+ */
+public interface GitHubConfigFileProvider {
+
+    /**
+     * Gets the configuration file at the given path from the main branch of the given repository,
+     * deserialized to the given type using Jackson.
```

Comment:
Not sure if we should mention this given you can also get a String. This can be useful for templates for instance.

Reply:
We have to mention it as Jackson annotations are the only way to control deserialization.

I'll also mention it's possible to retrieve the file as a String directly, though.